src/reader.rs

```diff
@@ -436,6 +528,27 @@ impl<R: Read + Seek> HeaderReader<R> {
         Ok(())
     }
 
+    fn reconstruct_geometry(&mut self, hierarchy: &mut (impl BufRead + Seek)) -> Result<()> {
```

Comment:
Can you add a short comment describing the rationale behind this function, please?
People who are less familiar with the FST format might be confused as to what is happening here.